### DISCUSSION TITLE
[iOS] unable to redirect to a specific screen when Notification caused the app to open from quit state

### DISCUSSION THREAD

**Discussion by TsedeyT** [invertase/react-native-firebase#4778] [Q&A]
```
//package.json
"dependencies": {
 "@firebase/firestore": "^2.1.1",
 "@react-native-firebase/app": "^10.3.0",
 "@react-native-firebase/firestore": "^10.3.1",
 "@react-native-firebase/messaging": "^10.3.1",
 "firebase": "^8.2.1",
 "react-native-bootsplash": "^3.1.1",
 "react-native": "0.63.3",
}


//FcmService.js
try {
      //when the application is opened from a quite state 
      messaging()
        .getInitialNotification()
        .then((remoteMessage) => {
          console.log(
            'Notification caused app to open from quit state:',
            remoteMessage,
          );
          if (remoteMessage) {
            onOpenNotification(remoteMessage);
          }
        })
        .catch((err) => console.log('Error: ', err));
    } catch (error) {
      console.log({error});
    }

//App.js
useEffect(() => {
fcmService.registerAppWithFCM();
fcmService.register(onRegister, onNotification, onOpenNotification);
function onRegister(token) {
      console.log('[App] onRegister: ', token);
    }
 function onNotification(notification) {
      console.log('[App] onNotification : ', notification);
 alert(
      "notification.data.title",
      "notification.data.message",
      [
        { text: "OK", onPress: () => console.log("OK Pressed") }
      ],
      { cancelable: false }
    );
}
 function onOpenNotification(notification) {
      console.log('[App] onOpenNotification: ', notification);

      const prefix = 'myApp://somescreen/';

      if (notification && notification.data) {
       
        const url = `${prefix}${notification.data.path}/`;
        Linking.openURL(url).catch((err) => console.error(err));
      }
    }
return () => {
    
      fcmService.unRegister();
     
    };
}
```

**Comment by mikehardy** [comment]
In my experience, separate the concerns. Have state storage somewhere that registers things, and consult that state on your welcome screen for optional forwarding after the App initial render.

Now when your app starts from a remote message, a notification open, or a dynamic link open, your registered handlers are called and store that they were called in state storage. Then your welcome screen forwards. 

This pattern works in my app

**Reply by TsedeyT** [reply]
@mikehardy do you have any example code?

**Reply by mikehardy** [reply]
No I do not. I use Resub which most people don't use so my code likely would not be interesting anyway. Redux is an option as well.

**Reply by TsedeyT** [reply]
> No I do not. I use Resub which most people don't use so my code likely would not be interesting anyway. Redux is an option as well.

when I use useRedux hook in App.js it is giving me an error and If I use notification register in the welcome screen, it doesn't solve the problem.  I haven't set Priorities to both Android and iOS...will that be an issue? how can I  do that?

**Reply by mikehardy** [reply]
There are so many ways to store state in react-native, and thus so many ways to do it wrong. I'm just telling you it is possible, and when I follow this pattern it works, but I can't trouble-shoot your project-specific code, sorry
